src/syntax/syntax.rkt

```diff
@@ -440,31 +480,48 @@
                         (define-values (_ exs) (real-apply compiler pt))
                         (if exs (first exs) fail))
                       (sym-append 'synth: name)))
-
   ;; Get floating-point implementation
   (define fl-proc
     (cond
-      [(assoc 'fl attrib-dict)
-       =>
-       cdr] ; user-provided implementation
-      [(operator-accelerator? op) ; Rival-synthesized accelerator implementation
+      [fl fl] ; user-provided implementation
+      [(operator-accelerator? new-op) ; Rival-synthesized accelerator implementation
        (match-define `(,(or 'lambda 'λ) (,vars ...) ,body) (operator-spec op-info))
        (synth-fl-impl name vars body)]
       [else ; Rival-synthesized operator implementation
        (define vars (build-list (length ireprs) (lambda (i) (string->symbol (format "x~a" i)))))
-       (synth-fl-impl name vars `(,op ,@vars))]))
+       (synth-fl-impl name vars `(,new-op ,@vars))]))
 
   ; update tables
-  (define impl (operator-impl name op-info ireprs orepr fl-proc))
+  (define impl (operator-impl name op-info (context vars orepr ireprs) spec fpcore fl-proc))
   (hash-set! operator-impls name impl)
-  (hash-update! operators-to-impls op (curry cons name)))
+  (hash-update! operators-to-impls new-op (curry cons name)))
 
-(define-syntax-rule (define-operator-impl (operator name atypes ...) rtype [key value] ...)
-  (register-operator-impl! 'operator
-                           'name
-                           (list (get-representation 'atypes) ...)
-                           (get-representation 'rtype)
-                           (list (cons 'key value) ...)))
+(define-syntax (define-operator-impl stx)
```

Comment:
This macro is a bit messy. Can we tidy it up? Maybe
```
(syntax-case stx (:)
  [(_ ...)
   (let ([id #'id] ... [fields #'(fields ...)])
     (unless (identifier? ...)
       ...)
     (define operator #f)
     ...
     (let loop ([fields fields])
       ...))])
```
More mutation. But I think it would be nicer. Also moves the identifier check higher up.